p2p/identity_service.go

```diff
@@ -0,0 +1,286 @@
+package p2p
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/helpers"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	protocol "github.com/libp2p/go-libp2p-core/protocol"
+	"github.com/qri-io/qri/config"
+	"github.com/qri-io/qri/event"
+	"github.com/qri-io/qri/repo"
+	"github.com/qri-io/qri/repo/profile"
+)
+
+const (
+	// ProfileProtocolID is the protocol id for the profile exchange service
+	ProfileProtocolID = protocol.ID("/qri/profile/0.1.0")
+	// ProfileTimeout is the length of time we will wait for a response in a
+	// profile exchange
+	ProfileTimeout = time.Minute * 2
+)
+
+var (
+	// expectedProtocols is the list of protocols we expect a node that has a
+	// profile service to speak
+	expectedProtocols = []string{
+		string(QriProtocolID),
```

Comment:
do we need the `QriProtocolID` service? I'd love to be able to deprecate it entirely.

Reply:
`QriProtocolID` is now the new `/qri/0.1.0` protocol id, not the old `/qri` protocol id.
My logic is it can be a protocol that we basically reserve to use, but probably won't ever use to send messages on, but rather as a way to potentially to control protocol versioning. If we see that the `QriProtocolID` is the wrong version, we can immediately back off or respond in a specific way

Reply:
again, not married to this idea, if you'd rather we remove it entirely, just a potential logic for keeping it

Reply:
love it. Let's keep it!